tfhe/src/core_crypto/algorithms/glwe_sample_extraction.rs

```diff
@@ -97,14 +97,14 @@ pub fn extract_lwe_sample_from_glwe_ciphertext<Scalar, InputCont, OutputCont>(
     InputCont: Container<Element = Scalar>,
     OutputCont: ContainerMut<Element = Scalar>,
 {
-    assert!(
-        input_glwe.glwe_size().to_glwe_dimension().0 * input_glwe.polynomial_size().0
-            == output_lwe.lwe_size().to_lwe_dimension().0,
-        "Mismatch between equivalent LweDimension of input ciphertext and output ciphertext. \
-        Got {:?} for input and {:?} for output.",
-        LweDimension(input_glwe.glwe_size().to_glwe_dimension().0 * input_glwe.polynomial_size().0),
-        output_lwe.lwe_size().to_lwe_dimension(),
-    );
+    let in_lwe_dim = input_glwe
+        .glwe_size()
+        .to_glwe_dimension()
+        .to_equivalent_lwe_dimension(input_glwe.polynomial_size());
+
+    let out_lwe_dim = output_lwe.lwe_size().to_lwe_dimension();
+
+    assert!(in_lwe_dim == out_lwe_dim);
```

Comment:
keep the message ? 😅 with just a very small modification to the original code ?

same elsewhere

Reply:
Done